components/director/internal/domain/apptemplate/converter_test.go

```diff
@@ -0,0 +1,135 @@
+package apptemplate_test
+
+import (
+	"database/sql"
+	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/apptemplate"
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConverter_ToEntity(t *testing.T) {
+	// given
+	id := "foo"
+	name := "bar"
+
+	appTemplateEntity := fixEntityAppTemplate(t, id, name)
+	appTemplateModel := fixModelAppTemplate(id, name)
+
+	testCases := []struct {
+		Name               string
+		Input              *model.ApplicationTemplate
+		Expected           *apptemplate.Entity
+		ExpectedErrMessage string
+	}{
+		{
+			Name:     "All properties given",
+			Input:    appTemplateModel,
+			Expected: appTemplateEntity,
+		},
+		{
+			Name:     "Empty",
+			Input:    &model.ApplicationTemplate{},
+			Expected: &apptemplate.Entity{},
+		},
+		{
+			Name:     "Nil",
+			Input:    nil,
+			Expected: nil,
+		},
+		// Cannot test error case while marshalling Application Input or Placeholders
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			conv := apptemplate.NewConverter()
+
+			// when
+			res, err := conv.ToEntity(testCase.Input)
+
+			if testCase.ExpectedErrMessage != "" {
+				require.Error(t, err)
+				assert.Equal(t, testCase.ExpectedErrMessage, err.Error())
+			} else {
+				require.Nil(t, err)
+			}
+
+			// then
+
+			assert.Equal(t, testCase.Expected, res)
+		})
+	}
+}
+
+func TestConverter_FromEntity(t *testing.T) {
+	// given
+	id := "foo"
+	name := "bar"
+
+	appTemplateEntity := fixEntityAppTemplate(t, id, name)
+	appTemplateModel := fixModelAppTemplate(id, name)
+
+	testCases := []struct {
+		Name               string
+		Input              *apptemplate.Entity
+		Expected           *model.ApplicationTemplate
+		ExpectedErrMessage string
+	}{
+		{
+			Name:               "All properties given",
+			Input:              appTemplateEntity,
+			Expected:           appTemplateModel,
+			ExpectedErrMessage: "",
+		},
+		{
+			Name:               "Empty",
+			Input:              &apptemplate.Entity{},
+			Expected:           &model.ApplicationTemplate{},
+			ExpectedErrMessage: "",
+		},
+		{
+			Name:               "Nil",
+			Input:              nil,
+			Expected:           nil,
+			ExpectedErrMessage: "",
+		},
+		{
+			Name: "Application Input Unmarshall Error",
+			Input: &apptemplate.Entity{
+				ApplicationInput: "{dasdd",
+			},
```

Comment:
I would add `Expected`: nil in both unmarshalling cases.

Reply:
Why? 

1) it is set automatically
2) when there is an error, I don't check value of the result, just error, as this is common approach in Go